osquery/tables/system/windows/drivers.cpp

```diff
@@ -173,79 +144,135 @@ Status getDeviceProperty(const device_infoset_t& infoset,
   return Status();
 }
 
-static inline std::string getDriverImagePath(const std::string& service_key) {
+std::string getDriverImagePath(const std::string& service_key) {
   SQL sql("SELECT data FROM registry WHERE path = '" + service_key +
           "\\ImagePath'");
-  if (sql.rows().size() == 1) {
-    auto path = sql.rows().at(0).at("data");
-    if (!path.empty()) {
-      CHAR systemRoot[MAX_PATH] = {0};
-      GetSystemDirectory(systemRoot, MAX_PATH);
-      return systemRoot +
-             boost::regex_replace(path, boost::regex("^.*[Ss]ystem32"), "");
-    }
+
+  if (sql.rows().empty() || sql.rows().at(0).count("data") == 0) {
+    return "";
   }
-  return "";
+
+  auto path = sql.rows().at(0).at("data");
+  if (path.empty()) {
+    return "";
+  }
+
+  char systemRoot[MAX_PATH] = {0};
+  GetSystemDirectory(systemRoot, MAX_PATH);
+  return systemRoot +
+         boost::regex_replace(path, boost::regex("^.*[Ss]ystem32"), "");
 }
 
 QueryData genDrivers(QueryContext& context) {
   QueryData results;
-
-  auto devInfoset = setupDevInfoSet();
+  auto devInfoset = setupDevInfoSet(DIGCF_ALLCLASSES | DIGCF_PRESENT);
   if (devInfoset == nullptr) {
     win32LogWARNING("Error getting device handle");
     return results;
   }
 
+  std::map<std::string, Row> apiDevices;
   std::vector<SP_DEVINFO_DATA> devices;
   auto ret = getDeviceList(devInfoset, devices);
   if (!ret.ok()) {
     win32LogWARNING(ret.getMessage(), ret.getCode());
     return results;
   }
 
+  // Then, leverage the Windows APIs to get whatever remains
   for (auto& device : devices) {
     char devId[MAX_DEVICE_ID_LEN] = {0};
     if (CM_Get_Device_ID(device.DevInst, devId, MAX_DEVICE_ID_LEN, 0) !=
         CR_SUCCESS) {
       win32LogWARNING("Failed to get device ID");
```

Comment:
Since you're at this, can we improve this error message with some device info like the device name or something?